### ISSUE TITLE
Validation Pipe cannot handle property with allowed null value

### ISSUE THREAD

**Issue by SeWaS** [nestjs/nest#2238]
## I'm submitting a...

<pre><code>
[X] Regression 
[ ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

After the hotfix for validation pipe (b3f51cc9c110017ebc8f6ec12da6b92d96f37caa) it seems that stripping prototypes has issues with optional values if they are set `null`. E.g.
```
class TestModel {

  @IsNumber()
  public prop: number;

  @IsOptional()
  @IsString()
  public optionalProp: string

}
```
with an object like
```
const obj = new TestModel();
obj.prop = 42;
obj.optionalProp = null;
```
seems to throw the following error:
```
TypeError: Cannot convert undefined or null to object
      at ValidationPipe.stripProtoKeys (packages/common/pipes/validation.pipe.ts:1:25200)
      at keys.filter.forEach.key (packages/common/pipes/validation.pipe.ts:1:25470)
      at Array.forEach (<anonymous>)
      at ValidationPipe.stripProtoKeys (packages/common/pipes/validation.pipe.ts:1:25400)
      .....
```

## Expected behavior
To work with allowed null values (defined by `class-validators` `@IsOptional()` decorator https://github.com/typestack/class-validator#validation-decorators)

## Minimal reproduction of the problem with instructions
I created a failing test in my forked repo: https://github.com/SeWaS/nest/commit/30ab8699b01a88d74d7c7d39aaf18b967b20be30

... and a suggestion for a solution:
https://github.com/SeWaS/nest/commit/4b73ca638eaa37e6e1de8454296f6d36988ab48c

## What is the motivation / use case for changing the behavior?
We have DTOs with some properties which are allowed to be `undefined`, `null` or of given type, whereby `undefined` and `null` have a different meaning for further processing.

## Environment

<pre><code>
Nest version: 6.2.3

 
For Tooling issues:
- Node version: 10.15.3
- Platform:  mac

Others:
- class-validator: 0.9.1
- class-transformer: 0.2.0

**Comment by kamilmysliwiec** [comment]
Thanks for reporting! Could you create a PR?

**Comment by kamilmysliwiec** [comment]
Published as 6.2.4 :)
